e3nn/math/normalize_activation.py

```diff
@@ -18,8 +18,17 @@ def moment(f, n, dtype=None, device=None):
 
 @compile_mode('trace')
 class normalize2mom(torch.nn.Module):
+    _is_id: bool
+
     def __init__(self, f, dtype=None, device=None):
         super().__init__()
+
+        # Try to infer a device:
+        if device is None and isinstance(f, torch.nn.Module):
+            # Avoid circular import
+            from e3nn.util._argtools import _get_device
+            device = _get_device(f)
+
         cst = 1 / moment(f, 2, dtype=dtype, device=device) ** 0.5
```

Comment:
If I remember well, before `cst` was a python float.
Isn't it simpler if cst is a float, no need to convert any type and so on?

Reply:
Switched back :+1: